tutor/src/store/courses.ts

```diff
@@ -0,0 +1,46 @@
+import { createSlice, createEntityAdapter, PayloadAction, current } from '@reduxjs/toolkit'
+import { useSelector } from 'react-redux'
+import { Course } from './types'
+import { updateCourse } from './api'
+import { bootstrap } from './bootstrap'
+
+const courseAdapter = createEntityAdapter<Course>({
+    sortComparer: (a, b) => a.name.localeCompare(b.name),
+})
+
+const initialState = courseAdapter.getInitialState()
+interface CourseSlice { courses: typeof initialState }
+
+const selectors = courseAdapter.getSelectors<CourseSlice>(s => s.courses)
+
+
+const coursesSlice = createSlice({
+    name: 'courses',
+    initialState,
+    reducers: {
+        rename(state, { payload: { id, name } }: PayloadAction<{ id: string, name: string }>) {
+            console.log(current(state))
+            const course = state.entities[id]
+            if (course) { course.name = name }
+            console.log(current(state))
+        }
+    },
+    extraReducers: (builder) => {
+        builder.addCase(updateCourse.fulfilled, (state, { payload: course }) => {
```

Comment:
I don't see any action created with `updateCourse.fulfilled`.

Reply:
the action is `updateCourse` from the `api` endpoints file: https://github.com/openstax/tutor-js/pull/3443/files/d4d8049b44fdbdc141672a7b4a104ac2b544df99

Maybe they should be put in the same file as the slice, but we've found it's better to keep all the urls together so you can find them by url.  I'd be fine either way really

Reply:
oh I see. I was thinking we could split the apis into different files, so it is not a massive file like the one we have now. I am ok adding it into the same file as the slice.

Reply:
ok. I got the `updateCourse.fulfilled`. It is from the `createAyncThunk`.